src/ops/fuelup_toolchain/export.rs

```diff
@@ -0,0 +1,322 @@
+
+        if component.name == component::FORC {
+            let forc_executables = component.executables;
+            for plugin in Components::collect_plugins()? {
+                if !forc_executables.contains(&plugin.name) {
+                    plugin.executables.iter().for_each(|executable| {
+                        let _ = exec_version(&export_toolchain.bin_path.join(executable))
+                            .map(|version| version_map.insert(executable.clone(), version));
+                    });
+                }
+            }
+        }
+    }
+    let mut export_channel = channel.unwrap_or(toolchain_name.clone());
+    if toolchain_override::Channel::from_str(&export_channel).is_err() {
+        println_warning(&format!(
+            "Invalid channel '{}', expected one of {}. \
+            \nPlease input a valid channel: ",
+            export_channel, VALID_CHANNEL_STR,
+        ));
+        let mut input_channel_name = String::new();
+        if reader.read_line(&mut input_channel_name).is_err() {
+            bail!("Failed to read user input");
+        }
+        input_channel_name = String::from(input_channel_name.trim());
+        if toolchain_override::Channel::from_str(&input_channel_name).is_err() {
+            bail!(
+                "Invalid channel '{}', expected one of {}.",
+                input_channel_name,
+                VALID_CHANNEL_STR,
+            );
+        } else {
+            export_channel = input_channel_name;
+        }
+    }
+
+    let toolchain_override = ToolchainOverride {
+        cfg: OverrideCfg::new(
+            ToolchainCfg {
+                // here shouldn't be err as we has checked above
+                channel: toolchain_override::Channel::from_str(&export_channel)?,
+            },
+            Some(version_map),
+        ),
+        path: toolchain_info_path.clone(),
+    };
+    let document = toolchain_override.to_toml();
+    if std::fs::write(toolchain_override.path, document.to_string()).is_err() {
+        bail!("Failed to write {}", toolchain_info_path.display());
+    }
+
+    info!(
+        "Exported toolchain \"{}\" to file: \"{}\"",
+        toolchain_name,
+        toolchain_info_path.display()
+    );
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        channel, path::settings_file, toolchain::DistToolchainName,
+        toolchain_override::ToolchainOverride,
+    };
+    use serial_test::serial;
+    use std::{fs, io::Write};
+
+    use super::*;
+
+    // simulate input
+    const INPUT_NOP: &[u8; 1] = b"\n";
+    const INPUT_YES: &[u8; 2] = b"y\r";
+    const INPUT_NO: &[u8; 4] = b"n \r\n";
+    const INPUT_INVALID_CHANNEL: &[u8; 11] = b"my-channel\n";
+    const INVALID_CHANNEL: &str = "my-channel";
+
+    fn remove_toolchain_info() {
+        if let Some(fuel_toolchain_toml_file) = get_fuel_toolchain_toml() {
+            if fuel_toolchain_toml_file.exists() {
+                fs::remove_file(fuel_toolchain_toml_file).unwrap();
```

Comment:
Do we want to change the `unwrap()`s to `expect(<msg>)` for clearer failures in tests or are we fine with unwrapping and crashing?